entity-services-examples/README.md

```diff
@@ -1,61 +1,81 @@
-
-Edit the filesystem output to be usful for your checkout, then execute it.
-
-
-### Step 4 deploy modified artifacts
-
-
-`gradle mlDeploy`
-
-
-### Step 5 Use them
-
-The QConsole workspace has just a couple of SQL and SPARQL queries at this point to peruse.
-
+* A QConsole workspace `races-qc.xml` contains XQuery scripting to work with
+ the Entity Services API.
+* Examples of code that has been generated, then edited are in directories under 
+ `src/main`
+* Java classes, with source under `src/main/java` demonstrate use of the Data
+ Movement SDK for an Entity Services toolchain.
+
+*Run all commands from this directory.  The example code will not deploy to
+MarkLogic if run from the parent project's directory.*
+
+Step One. Deploy
+-----------------
+
+`../gradlew mlDeploy` deploys code to a MarkLogic server.  To change the
+location of the deployment edit `gradle.properties`
+
+* Databases and forests are provisioned as configured in `src/main/ml-config`.
+ Entity services generated the database configuration.
+* Entity Services code modules to go from `src/main/ml-modules/ext` to the
+ REST extension space.  This code was created from stubs generated by Entity 
+ Services.
+* Transforms go from `src/main/ml-modules/transforms` to REST config.  These
+ transforms wrap calls to the Entity Services modules, but were authored by
+ a developer.
+* Search options are installed from `src/main/ml-modules/options` to REST.
+ This configuration was generated by Entity Services.
+* A module to generate code is installed from
+ `src/main/ml-modules/services` to REST.  This extension is just used by the
+ example ode to invoke the Entity Services API as a demonstration.
+* A schema that can validate canonical instance data is installed from
+ `src/main/ml-schemas` to the schemas database.  Entity Services API generated
+ this schema out-of-the-box.
+* An extraction template that indexes canonical instance data is installed
+ from `src/main/ml-schemas` to the schemas database in the proper TDE
+ collection.  It is a somewhat modified version of one generated by Entity
+ Services.
+
+Step Two.  Run
+--------------
+
+`../gradlew run` runs the examples loader.  If you changed deployment
+information, check `src/main/resources/application.properties` and edit as
+required.
+
+* It loads documents "as-is" from the `data` directory.  When this step is
+ complete, the database will have lots of JSON documents as staged data.
+* It loads some RDF reference data.
+* It invokes an in-place transform batch on the raw documents.  This transform
+ invokes code generated by Entity Services to wrap the JSON documents and store
+ them in their canonical format.
+* Lastly, it actually invokes the code generators and saves the results to the
+ `gen` directory.  Note that this step is conceptually distinct from the
+ previous ones, and would have had to already happen in a typical scenario.
```

Comment:
I didnt understand 2 things here:

- by "previous ones" do you mean bullets in step2?
- what is meant by the last sentence "would have had to already happen in a typical scenario"